src/main/java/net/mcreator/preferences/PreferencesManager.java

```diff
@@ -18,44 +19,214 @@
-				storePreferences(new PreferencesData());
+				loadedPreferences = gson.fromJson(FileIO.readFileToString(file), Map.class);
+			} catch (JsonSyntaxException e) {
+				LOG.error("The preferences file could not be loaded. Default values will be used.", e);
 			}
+			loadPreferences("mcreator");
+		}
+	}
+
+	/**
+	 * <p>Once plugins are loaded, we can now load preferences registered by them with {@link ApplicationLoadedEvent}.</p>
+	 */
+	public static void initNonCore() {
+		if (!firstLoading) {
+			PREFERENCES_REGISTRY.forEach((identifier, preferences) -> {
+				if (!identifier.equals("mcreator"))
+					loadPreferences(identifier);
+			});
+		} else {
+			savePreferences(); // if this is the first time, we save preferences for the next time
 		}
 	}
 
-	public static void storePreferences(PreferencesData data) {
-		PREFERENCES = data;
-		FileIO.writeStringToFile(gson.toJson(data), preferencesFile);
+	/**
+	 * <p>Load all preferences registered under the provided {@param identifier}.</p>
+	 *
+	 * @param identifier <p> The identifier used to get the preference file to load</p>
+	 */
+	public static void loadPreferences(String identifier) {
+		try {
+			LOG.debug("Loading preferences from " + identifier);
+			if (loadedPreferences.containsKey(identifier)) {
+				// Convert values from the file to properly work
+				loadedPreferences.get(identifier).forEach((category, entries) -> entries.forEach(
+						entry -> getPreferencesRegistry().get(identifier).stream()
+								.filter(preference -> preference.getID().equals(entry.get("id")))
+								.forEach(preference -> {
+									if (preference.get() instanceof Locale)
+										preference.set(
+												Locale.forLanguageTag(((String) entry.get("value")).replace("_", "-")));
+									else if (preference.get() instanceof Color)
+										preference.set(new Color((int) (double) entry.get("value")));
+									else if (entry.get("value") instanceof Double d)
+										preference.set(d.intValue());
+									else
+										preference.set(entry.get("value"));
+								})));
+			} else {
+				LOG.debug(identifier + " has no saved values. Default values will be used.");
+			}
+		} catch (Exception e) {
+			LOG.error("Failed to load preferences. Reloading defaults.", e);
+			resetSpecific(identifier);
+		}
+	}
+
+	/**
+	 * <p>Save preferences of all identifiers registered inside PREFERENCES_REGISTRY.</p>
+	 */
+	public static void savePreferences() {
+		Map<String, Map<String, List<PreferencesEntry<?>>>> allPreferences = new HashMap<>();
+
+		PREFERENCES_REGISTRY.forEach((identifier, preferences) -> {
+			Map<String, List<PreferencesEntry<?>>> identifierPrefs = new HashMap<>();
+			preferences.forEach(entry -> {
+				// We check if the section doesn't exist to add it
+				if (!identifierPrefs.containsKey(entry.getSectionKey()))
+					identifierPrefs.put(entry.getSectionKey(), new ArrayList<>());
+
+				// We change the registered value for some types, so we can load them correctly

```

Comment:
Why is this needed? GSON is well capable of serializing Colors and Locales

Reply:
It is able to serialize them, but the problem is when we try to read those values and set the preference entries. Values don't have the same/correct format, so they are reset.

Reply:
I don't know. I simply know that every time I tested without them, it never worked. (Concerning the language thing, this is due to the format used when serializing, it's not the same format as the one used when creating a new Locale object for some reasons)

Reply:
Well this workaround is not ok as it hardcodes things, where this PR aims to add support for any entry type, meaning the user may want to serialize other objects and this is not possible if serialization/deserialization is handled the way it is right now

Reply:
> Well this workaround is not ok as it hardcodes things, where this PR aims to add support for any entry type, meaning the user may want to serialize other objects and this is not possible if serialization/deserialization is handled the way it is right now
> 
> > I don't know.
> 
> Each entry could return class it uses for serialization which is then passed to the GSON

It actually tries, but due to the preferences dualif, it can't. It opens most of things, but the usable entry types are still limited to the implemented ones (because each JComponent uses a different method to get their value)